### PR TITLE
feat: update dependencies

### LINKED ISSUE #19: Upgrade dependencies
Currently, the dependencies are a bit stale, specifically bloc is 2 major versions behind.  It would be nice to have these updated to be able to use newer features such as Cubit, which simplifies state management.

### PR DESCRIPTION
**Pull request by chimon2000**
Closes #19, #22 

- [x] Update dependencies
- [x] Update bloc code
- [x] Add unit tests

### CODE REVIEW

**Comment by chimon2000** [comment]
@leonardo2204 I only added a few tests to get the ball rolling.

**Comment by leonardo2204** [comment]
Hey @chimon2000 ,thanks for your effort! Later the Day I'll check your PR!

File: test/blocs/bottom_bar/bottom_bar_bloc_test.dart [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    tearDown(() {
+      bottomBarBloc?.close();
+    });
+
+    test('initial state.selectedIndex is 0', () {
+      expect(bottomBarBloc.state.selectedIndex, 0);
+    });
+
+    blocTest(
+      'emits [1] when CounterEvent.increment is added',
```
Comment (leonardo2204) [inline test/blocs/bottom_bar/bottom_bar_bloc_test.dart]: does this text make sense here?

**Review by leonardo2204 — APPROVED** [review]

**Comment by leonardo2204** [comment]
LGTM, thanks for the effort!!

File: test/blocs/bottom_bar/bottom_bar_bloc_test.dart (hunk 1 above)
Comment (chimon2000) [reply test/blocs/bottom_bar/bottom_bar_bloc_test.dart]: Nope, will update!

**Comment by chimon2000** [comment]
Added one more fix for #22 which I discovered while trying to run locally.

**Comment by leonardo2204** [comment]
Thanks! Feel free to merge it :)

**Comment by chimon2000** [comment]
@leonardo2204 I don't think I have permissions to merge.

**Comment by leonardo2204** [comment]
Sorry, just merged it! Thanks again!

**Pull request merged**